nickhartdev/Overlook
# Fetch setup

###### What's this PR do?
- adds a dataFetcher file, where all fetch API calls will live
- sets up the start of a DOM manipulator object - used in this case to play with how I chose to fetch the data
- adds another conditional to validateDataType for more thorough error handling
###### Where should the reviewer start?
data-fetcher.js, to see where it all starts.
###### How should this be manually tested?
Run the code in the browser, you should see a user name and list of bookings objects show up.
###### Any background context you want to provide?
This became a more experimental phase of the project - I messed around a lot with where and how the fetches were called to figure out what would be optimal. The start of DOMUpdates.js and small modifications made in index.html and index.js were there to help me figure out how I would want to display my information and translate the methods called from dataFetcher.
###### What are the relevant tickets?
Closes #32, progress on #31
